src/Nanoid/Nanoid.cs

```diff
@@ -1,107 +1,278 @@
-        /// <param name="random"></param>
-        /// <param name="alphabet"></param>
-        /// <param name="size"></param>
-        /// <returns></returns>
-        /// <exception cref="ArgumentNullException"></exception>
-        /// <exception cref="ArgumentOutOfRangeException"></exception>
-        public static string Generate(Random random, string alphabet = DefaultAlphabet, int size = 21)
+        /// <param name="alphabet">The set of characters used in generating the id. Defaults to <see cref="Alphabets.Default">Alphabets.Default.</see></param>
+        /// <param name="size">The length of the id. Defaults to <see cref="DefaultIdSize"/>.</param>
+        /// <returns>A new string representing a random nanoid with the specified <paramref name="alphabet"/> and <paramref name="size"/>.</returns>
+        /// <exception cref="ArgumentNullException">If any of the provided arguments are null.</exception>
+        /// <exception cref="ArgumentOutOfRangeException">If <paramref name="alphabet"/>'s length is outside the range [0, 256] or if <paramref name="size"/> is less than or equal to 0.</exception>
+        public static string Generate(string alphabet = Alphabets.Default, int size = 21)
         {
             Validate(alphabet, size);
-            return GenerateId(random, alphabet, size);
+            return GenerateImpl(GlobalRandom, alphabet, size);
         }
 
-        private static void Validate(string alphabet, int size)
+        /// <summary>
+        /// Generate a Nanoid.
+        /// </summary>
+        /// <param name="random">A random number generator.</param>
+        /// <param name="alphabet">The set of characters used in generating the id. Defaults to <see cref="Alphabets.Default">Alphabets.Default.</see></param>
+        /// <param name="size">The length of the id. Defaults to <see cref="DefaultIdSize"/>.</param>
+        /// <returns>A new string representing a random nanoid with the specified <paramref name="alphabet"/> and <paramref name="size"/>.</returns>
+        /// <exception cref="ArgumentNullException">If any of the provided arguments are null.</exception>
+        /// <exception cref="ArgumentOutOfRangeException">If <paramref name="alphabet"/>'s length is outside the range [0, 256] or if <paramref name="size"/> is less than or equal to 0.</exception>
+        public static string Generate(Random random, string alphabet = Alphabets.Default, int size = 21)
         {
-            if (Random == null)
-            {
-                throw new ArgumentNullException("random cannot be null.");
-            }
+            Validate(alphabet, size);
+            return GenerateImpl(random, alphabet, size);
+        }
 
+        private static void Validate(string alphabet, int size)
+        {
             if (alphabet == null)
             {
-                throw new ArgumentNullException("alphabet cannot be null.");
+                throw new ArgumentNullException(nameof(alphabet), "alphabet cannot be null.");

```

Comment:
I got warnings that the recommended way to instantiate exceptions like this one is to pass `nameof(argument)` as the first parameter so did so here.

Not sure if the message "alphabet cannot be null" is needed anymore tho.